app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,9 @@
+class CommentsController < ApplicationController
+  def create
+    @user = current_user
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new(user: @user, post: @post, text: params[:text])
+    @comment.save
+    redirect_to user_post_path(@user, @post)
+  end
+end
```

Comment:
- [ ] Great job so far 👏🏻 However according to the project requirements, The comments controller should have `new` and `create` actions similar to the one you have created inside the CommentsController, Kindly add the `new` action to your comments controller It might be some like this:


```ruby
# controllers/comments_controller.rb

class CommentsController < ApplicationController
  ...

  def new
    @comment = Comment.new
  end
  def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      redirect_to post_path(@post)
    end
  end

  private 
  def comment_params
    params.require(:comment).permit(:text)
  end

	...
end
```